inference/models/inference.py

```diff
@@ -0,0 +1,292 @@
+from django.db import models
+from django.utils import timezone
+
+from inference.models import ClassificationType, ExternalJobStatus, InferenceJobStatus
+from inference.utils.batch import BatchConfig, BatchProcessor
+from inference.utils.inference_api_client import InferenceAPIClient, ModelManager
+
+
+class ModelVersion(models.Model):
+    """
+    Allows us to maintain tracking between multiple versions of a classification model.
+    """
+
+    api_identifier = models.CharField(max_length=255)
+    description = models.TextField()
+    classification_type = models.IntegerField(
+        choices=ClassificationType.choices, help_text="Type of classification this model performs"
+    )
+    is_active = models.BooleanField(
+        default=True, help_text="Whether this is the current active version for its classification type"
+    )
+
+    class Meta:
+        constraints = [
+            models.UniqueConstraint(
+                fields=["classification_type", "is_active"],
+                condition=models.Q(is_active=True),
+                name="unique_active_version",
+            )
+        ]
+
+    def __str__(self):
+        return f"{self.get_classification_type_display()} - {self.api_identifier}"
+
+    @classmethod
+    def get_active_version(cls, classification_type: int) -> "ModelVersion":
+        """Get the current active model version for a classification type."""
+        return cls.objects.get(classification_type=classification_type, is_active=True)
+
+    def set_as_active(self):
+        """Set this version as the active one for its classification type."""
+        # Deactivate other versions of this classification type
+        ModelVersion.objects.filter(classification_type=self.classification_type, is_active=True).exclude(
+            id=self.id
+        ).update(is_active=False)
+
+        # Set this one as active
+        self.is_active = True
+        self.save()
+
+
+class InferenceJob(models.Model):
+    """
+    Tracks an inference job for a collection of URLs.
+    One InferenceJob can have multiple ExternalJobs (one per batch).
+    """
+
+    collection = models.ForeignKey(
+        "sde_collections.Collection", on_delete=models.CASCADE, related_name="inference_jobs"
+    )
+    model_version = models.ForeignKey(
+        ModelVersion,
+        on_delete=models.PROTECT,  # Prevent deletion of ModelVersions that have associated jobs
+        related_name="inference_jobs",
+    )
+
+    created_at = models.DateTimeField(default=timezone.now)
+    updated_at = models.DateTimeField(auto_now=True)
+    completed_at = models.DateTimeField(null=True, blank=True)
+
+    status = models.IntegerField(choices=InferenceJobStatus.choices, default=InferenceJobStatus.QUEUED)
+    error_message = models.TextField(blank=True)
+
+    class Meta:
+        ordering = ["-created_at"]
+        constraints = [
+            models.UniqueConstraint(
+                fields=["collection", "model_version"],
```

Comment:
This needs to specifically be model_version.model_type, two versions with the same type will be attempting to overwrite the same field.